src/literal.h

```diff
@@ -33,52 +33,69 @@ class Literals;
 struct ExceptionPackage;
 
 class Literal {
+public:
+  // Type of the literal. Immutable because the literal's payload depends on it.
+  const Type type;
```

Comment:
Initially suspected that the memory corruption problem above might have something to do with other code changing the type of a literal holding an exception package, so I made the type immutable to have a safeguard.